templates/core/terraform/state-store/statestore.tf

```diff
@@ -25,3 +25,34 @@ resource "azurerm_cosmosdb_sql_database" "tre-db" {
   account_name        = azurerm_cosmosdb_account.tre-db-account.name
   throughput          = 400
 }
+
+resource "azurerm_private_dns_zone" "statestore" {
```

Comment:
The DNS zone resource isn't necessarily specific to statestore - it's all cosmos. Can the name reflect that?

Reply:
Do you think that is a better name? statestore is cosmos in our domain and hence isn't it better to be more concrete than being more general.

Reply:
I guess it's more convention. Becomes more of an issue when have multiple resources sharing the same DNS zone. So might have multiple web apps sharing a azure website private link DNS zone, so would call it `azurewebsites_private_dns_zone`, rather than `management_api_dns_zone`. 

Are there any DNS zones already in the templates?

Reply:
Cool, thanks!